unitTests/ui/DynamicFacet/DynamicFacetValues/DynamicRangeFacetValueCreatorTest.ts

```diff
@@ -0,0 +1,178 @@
+      should format the value from the range properly`, () => {
+        responseValue.endInclusive = false;
+        initializeValue();
+        expect(facetValue.value).toBe(`${responseValue.start}..${responseValue.end}${RangeEndScope.Exclusive}`);
+      });
+
+      it('should have the formated displayValue from the value by default', () => {
+        expect(facetValue.displayValue).toBe(`${facetValue.start} ${facet.options.valueSeparator} ${facetValue.end}`);
+      });
+
+      it(`when is was a previous corresponding facet value
+      should copy it's displayValue`, () => {
+        const displayValue = 'this is a display value';
+        const correspondingFacetValue = facet.values.get(facetValue);
+        correspondingFacetValue.displayValue = displayValue;
+
+        initializeValue();
+        expect(facetValue.displayValue).toBe(displayValue);
+      });
+    });
+
+    describe('testing createFromRanges', () => {
+      let index: number;
+      let rangeValue: IRangeValue;
+      let facetValue: DynamicFacetValue;
+
+      beforeEach(() => {
+        index = 0;
+        rangeValue = DynamicRangeFacetTestUtils.createFakeRanges(1)[0];
+        initializeValue();
+      });
+
+      function initializeValue() {
+        facetValue = valueCreator.createFromRange(rangeValue, index);
+      }
+
+      it('should have the right basic properties', () => {
+        expect(facetValue.start).toBe(rangeValue.start);
+        expect(facetValue.endInclusive).toBe(rangeValue.endInclusive);
+        expect(facetValue.end).toBe(rangeValue.end);
+        expect(facetValue.state).toBe(FacetValueState.idle);
+        expect(facetValue.position).toBe(index + 1);
+        expect(facetValue.numberOfResults).toBe(0);
+      });
+
+      it(`when the endInclusive is false
+      should format the value from the range properly`, () => {
+        expect(facetValue.value).toBe(`${facetValue.start}..${facetValue.end}${RangeEndScope.Exclusive}`);
+      });
+
+      it(`when the endInclusive is true
+      should format the value from the range properly`, () => {
+        rangeValue.endInclusive = true;
+        initializeValue();
+        expect(facetValue.value).toBe(`${facetValue.start}..${facetValue.end}${RangeEndScope.Inclusive}`);
+      });
+
+      it(`when there is no label 
+        should format the displayValue correctly`, () => {
+        expect(facetValue.displayValue).toBe(`${facetValue.start} ${facet.options.valueSeparator} ${facetValue.end}`);
+      });
+
+      it(`when there is a label 
+        should assign it to the displayValue`, () => {
+        const label = 'this is a label';
+        rangeValue.label = label;
+        initializeValue();
+        expect(facetValue.displayValue).toBe(label);
+      });
+    });
+
+    describe('testing createFromValue', () => {
+      let value = `10..20${RangeEndScope.Inclusive}`;
+      let facetValue: DynamicFacetValue;
+
+      beforeEach(() => {
+        initializeValue();
+      });
```

Comment:
Same as my other comment: should tests that don't run `initializeValue()` on their own be moved into another `describe` along with this `beforeEach`?